Assignment 5.Rmd

```diff
@@ -40,42 +38,55 @@ We want to see if we can build a decision tree to help teachers decide which stu
 
 #Visualize our outcome variable "score"
 ```{r}
-
+hist(D1$score)
 ```
 
 #Create a categorical outcome variable based on student score to advise the teacher using an "ifelse" statement
 ```{r}
-D1$advice <- 
+sort(D1$score)[126]
+sort(D1$score)[126+126]
+D1$advice <- ifelse(D1$score < 0.5, "intervene", ifelse(D1$score < 0.84, "monitor", "no action"))
 ```
 
 #Build a decision tree that predicts "advice" based on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required
 ```{r}
-score_ctree <- 
+score_ctree <- ctree(factor(advice) ~ prior_prob_count + prior_percent_correct + hints, D1)
 ```
 
 #Plot tree
 ```{r}
-
+plot(score_ctree)
 ```
 
 Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
 
+when students ask for more than 22 hints, they are more likely to need teachers' intervention.
+
+
+
 #Test Tree
 Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
 
 ```{r}
 #Upload new data
 
-D2 <- 
+D2 <- read.csv("intelligent_tutor_new.csv", header = T)
 
 #Generate predicted advice using the predict() command for new students based on tree generated from old students
 
-D2$prediction <- 
+D2$prediction <- predict(score_ctree, D2)
 
 ``` 
 ## Part III
 Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
 
+```{r}
+# all students have a score of 1, so they do not need action.
+# therefore, the accuracy of the model should be
+accuracy <- mean (ifelse(D2$prediction == "no action", 1,0))
+```
```

Comment:
Overall, great job on the coding part of this assignment but you may want to work on the analysis. As in, specify whether the model overfits or whether it is more generalizable in terms of its accuracy rate.